packages/invoices/src/OutboundShipment/api/hooks.ts

```diff
@@ -317,33 +311,19 @@ export const useDeleteSelectedLines = (): {
 } => {
   const { success, info } = useNotification();
   const { items, lines } = useOutboundRows();
-  const { mutate } = useDeleteInboundLine();
+  const { mutate } = useDeleteInboundLines();
   const t = useTranslation('distribution');
 
-  const { selectedRows } = useTableStore(state => {
+  const selectedRows = useTableStore(state => {
     const { isGrouped } = state;
 
     if (isGrouped) {
-      return {
-        selectedRows: (
-          Object.keys(state.rowState)
-            .filter(id => state.rowState[id]?.isSelected)
-            .map(selectedId => items?.find(({ id }) => selectedId === id))
-            .filter(Boolean) as OutboundItem[]
-        )
-          .map(({ lines }) => lines)
-          .flat()
-          .map(({ id }) => id),
-      };
+      return items
+        ?.filter(({ id }) => state.rowState[id]?.isSelected)
+        .map(({ lines }) => lines.flat())
+        .flat();
     } else {
-      return {
-        selectedRows: (
-          Object.keys(state.rowState)
-            .filter(id => state.rowState[id]?.isSelected)
-            .map(selectedId => lines.find(({ id }) => selectedId === id))
-            .filter(Boolean) as OutboundLineFragment[]
-        ).map(({ id }) => id),
-      };
+      return lines.filter(({ id }) => state.rowState[id]?.isSelected);
```

Comment:
this seems a little nicer!

Reply:
much 😄 
Might almost be small enough to 

```
    return isGrouped
      ? items
          ?.filter(({ id }) => state.rowState[id]?.isSelected)
          .map(({ lines }) => lines.flat())
          .flat()
      : lines.filter(({ id }) => state.rowState[id]?.isSelected);
```

Reply:
Nice! I see the suggestion in another PR also, will do both at once